R/position-jitterdodge.R

```diff
@@ -35,17 +35,21 @@ PositionJitterdodge <- ggproto("PositionJitterdodge", Position,
   jitter.height = NULL,
   dodge.width = NULL,
 
-  required_aes = c("x", "y", "fill"),
+  required_aes = c("x", "y"),
 
   setup_params = function(self, data) {
     width <- self$jitter.width %||% resolution(data$x, zero = FALSE) * 0.4
-    # Adjust the x transformation based on the number of 'fill' variables
-    nfill <- length(levels(data$fill))
-
+    # Adjust the x transformation based on the number of 'dodge' variables
+    dodgecols <- intersect(c("fill", "colour", "linetype", "shape", "size", "alpha"), colnames(data))  
+    if (length(dodgecols) == 0) stop("`position_jitterdodge()` requires ",
```

Comment:
It would be more natural to break the line just before `stop()`, and I prefer to always put if statements in `{}`